lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -509,6 +509,49 @@ static CallExpr *createContainerKeyedByCall(ASTContext &C, DeclContext *DC,
                                   C.AllocateCopy(argLabels));
 }
 
+/// Looks up the property corresponding to the indicated coding key. Returns a
+/// tuple containing the proeprty, the type that should be passed to the decoding method, and
+/// a boolean indicating whether `*IfPresent` method variants should be used.
+static std::tuple<VarDecl *, Type, bool>
+lookupVarDeclForCodingKeysCase(DeclContext *conformanceDC,
+                               EnumElementDecl *elt,
+                               NominalTypeDecl *targetDecl) {
+  ASTContext &C = elt->getASTContext();
+
+  for (auto decl : targetDecl->lookupDirect(DeclName(elt->getName()))) {
+    if (auto *vd = dyn_cast<VarDecl>(decl)) {
+      if (!vd->isStatic()) {
+        // This is the VarDecl we're looking for.
+
+        auto varType = conformanceDC->mapTypeIntoContext(vd->getInterfaceType());
+
+        if (auto referenceType = varType->getAs<ReferenceStorageType>()) {
```

Comment:
You can do `varType-> getValueInterfaceType()` and get rid of this if check.

Reply:
Thanks for the tip! I'm testing that change now.